src/IO/ReadSettings.h

```diff
@@ -97,22 +97,8 @@ struct ReadSettings
     std::optional<FileCacheUserInfo> filecache_user_info;
     bool enable_hdfs_pread = true;
 
-    ReadSettings adjustBufferSize(size_t file_size) const
-    {
-        ReadSettings res = *this;
-        res.local_fs_buffer_size = std::min(std::max(1ul, file_size), local_fs_buffer_size);
-        res.remote_fs_buffer_size = std::min(std::max(1ul, file_size), remote_fs_buffer_size);
-        res.prefetch_buffer_size = std::min(std::max(1ul, file_size), prefetch_buffer_size);
```

Comment:
Seems not adjuting `prefetch_buffer_size` here breaks this logic https://github.com/ClickHouse/ClickHouse/blob/6d89227370982c81abe50a170b6bbfef1b5bfc9e/src/Storages/MergeTree/MergeTreePrefetchedReadPool.cpp#L347-L355